src/lib/wrapAISDK.ts

```diff
@@ -0,0 +1,240 @@
+    const otlpExporter = new OTLPTraceExporter({
+      url: tracingUrl,
+      headers:
+        apiKey ?
+          {
+            Authorization: `Bearer ${apiKey}`,
+          }
+        : {},
+    });
+
+    // Add batch span processor with OTLP exporter
+    spanProcessors.push(
+      new BatchSpanProcessor(otlpExporter, {
+        maxExportBatchSize,
+      }),
+    );
+  } catch (error) {
+    // Handle initialization errors
+    console.error('Failed to initialize Scorecard telemetry:', error);
+  }
+
+  // Create tracer provider with span processors
+  tracerProvider = new NodeTracerProvider({
+    resource,
+    spanProcessors,
+  });
+
+  // Register the provider
+  tracerProvider.register();
+
+  // Get tracer instance
+  tracer = tracerProvider.getTracer('ai-sdk-wrapper');
+
+  return tracer;
+}
+
+/**
+ * Wraps the AI SDK module to automatically inject telemetry configuration
+ *
+ * @param aiSDKModule - The AI SDK module (e.g., import('ai'))
+ * @param config - Optional Scorecard configuration
+ * @returns Proxied AI SDK module with automatic telemetry injection
+ *
+ * @example
+ * ```typescript
+ * import ai from 'ai';
+ * import { wrapAISDK } from 'scorecard-ai/lib/wrapAISDK';
+ *
+ * const aiSDK = await wrapAISDK(ai);
+ *
+ * // Now all AI SDK calls will automatically send traces to Scorecard
+ * const { text } = await aiSDK.generateText({
+ *   model: openai('gpt-4o-mini'),
+ *   prompt: 'What is the capital of France? Answer in one sentence.',
+ * });
+ * ```
+ */
+export function wrapAISDK<T extends Record<string, unknown>>(
+  aiSDKModule: T,
+  config: ScorecardConfig = {},
+): T {
+  const projectId = config.projectId || readEnv('SCORECARD_PROJECT_ID');
+  const apiKey = config.apiKey || readEnv('SCORECARD_API_KEY');
+  const serviceName = config.serviceName || DEFAULT_SERVICE_NAME;
+  if (!apiKey) {
+    throw new Error('SCORECARD_API_KEY environment variable is not set');
+  }
+  const client = new Scorecard({ apiKey });
+
+  if (config.metrics && config.metrics.length > 0 && !projectId) {
+    throw new ScorecardError(
+      "The SCORECARD_PROJECT_ID environment variable is missing or empty; either provide it, or instantiate the AI SDK wrapper with a projectId option, like ScorecardAIWrapper({ projectId: '123' }).",
+    );
+  }
+
+  // Create metrics and monitor if needed
+  if (config.metrics && config.metrics.length > 0 && projectId) {
+    client
+      .put(`/api/v2/projects/${projectId}/monitors`, {
+        body: {
+          description: serviceName,
+          metrics: config.metrics,
+          filter: {
+            serviceName,
+          },
+        },
+      })
+      // we use .catch instead of try/catch to avoid needing async/await in the wrapper
+      // otherwise the code would await here before giving back the wrapped AI SDK module
+      .catch((error) => {
+        console.error(
+          `Failed to create a monitor for your traces. The most common case is your projectId is incorrect.`,
```

Comment:
optional nit: explicitly mentioning "Scorecard" makes the error more immediately clear:

```suggestion
          'Failed to create a Scorecard monitor for your AI SDK traces. The most common case is that the projectId is incorrect.',
```